### PR TITLE
WIP: Doc updates

### PR DESCRIPTION
**Pull request by martindurant**
fleshing out the docs

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -3,19 +3,60 @@ S3Fs
...
+allows typical file-system style operations like `cp`, `mv`, `ls`, `du`, `glob`,
+etc., as well as put/get of local files to/from S3.
+
+The connection can be anonymous - in which case only publicly-available,
+read-only buckets are accessible - or via credentials explicitly supplied
+or in configuration files.
+
+Calling `open()` on a `S3FileSystem` (typically using a context manager)
+provides an `S3File` for read or write access to a particular key. The
+object emulated the standard `File` protocol (`read`, `write`, `tell`, `seek`),
```
Comment (mrocklin) [inline docs/source/index.rst]: emulates, present tense.

File: docs/source/index.rst [hunk 2]
```diff
@@ -3,19 +3,60 @@ S3Fs
...
+such that functions expecting a file can access S3. Only binary read and write
+modes are implemented, with blocked caching.
+
+This project was originally designed as a storage-layer interface
+for `dask.distributed`.
```
Comment (mrocklin) [inline docs/source/index.rst]: Link

```
for dask.distributed_

.. _dask.distributed: https://distributed.readthedocs.org/en/latest
```

File: docs/source/index.rst [hunk 3]
```diff
@@ -3,19 +3,60 @@ S3Fs
...
    ...     print(f.read())
    b'Hello, world'
 
+(see also `walk` and `glob`)
+
+Reading with delimited blocks:
+
+.. code-block:: python
+
+   >>> s3.read_block(path, 1000, 1010, b'\n')
```
Comment (mrocklin) [inline docs/source/index.rst]: Suggest the use of keywords here to be more explicit, `start=`, `stop=`, `delimiter=`.

File: docs/source/index.rst [hunk 4]
```diff
@@ -3,19 +3,60 @@ S3Fs
...
+.. code-block:: python
+
+   >>> s3 = s3fs.S3FileSystme(anon=False)  # uses default credentials
+   >>> with s3.open('mybucket/new-file', 'wb') as f:
+   ...     f.write(2*2**20 * b'a')
+   ...     f.write(2*2**20 * b'a')
+   ...     f.write(2*2**20 * b'a') # buffer now >5MB, auto-flush here
+   ...     f.write(2*2**20 * b'a') # last write, data is flushed and file closed
+   >>> s3.du('mybucket/new-file')
+   {'MDtemp/new-file': 8388608}
```
Comment (mrocklin) [inline docs/source/index.rst]: We might not want to expose the details of blocked caching in the frontpage.  I suspect that this interests us more than the casual user.

File: docs/source/index.rst (hunk 3 above)
Comment (martindurant) [inline docs/source/index.rst]: *Excellent* suggestion - should have been start, length :)

File: docs/source/index.rst [hunk 5]
```diff
@@ -3,19 +3,70 @@ S3Fs
...
+   {'mybucket/new-file': 4194304}
+
+Limitations
+-----------
+
+This project is meant for convenience, rather than feature completeness.
+The following are known current omissions:
+- there is no append-mode on files
+- file access is always binary
+- no permissions/access-control (i.e., no chmod/chmown methods)
```
Comment (mrocklin) [inline docs/source/index.rst]: +1 on explicitly citing limitations like these.

Comment (mrocklin) [inline docs/source/index.rst]: Might want to check that the docs build nicely locally (not sure if you're doing this already or not)

```
pip install sphinx numpydoc
cd docs
make html
open build/html/index.html
```

**Comment by mrocklin** [comment]
+1

**Pull request merged**